src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -30,21 +36,29 @@ public class DocumentDatabase : IResourceStore
         private readonly CancellationTokenSource _databaseShutdown = new CancellationTokenSource();
         public readonly PatchDocument Patch;
 
-		private readonly HttpJsonRequestFactory _httpJsonRequestFactory = new HttpJsonRequestFactory(16);
+		private HttpJsonRequestFactory _httpJsonRequestFactory;
 
         private readonly object _idleLocker = new object();
         private Task _indexStoreTask;
         public bool LazyTransactionMode { get; set; }
         public DateTime LazyTransactionExpiration { get; set; }
         public TransactionOperationsMerger TxMerger;
+	    private DocumentConvention _convention;
 
-        public DocumentDatabase(string name, RavenConfiguration configuration, MetricsScheduler metricsScheduler,
+	    public string Url { get; private set; }
+
+	    public DocumentDatabase(string name, RavenConfiguration configuration, MetricsScheduler metricsScheduler,
             LoggerSetup loggerSetup)
         {
             Name = name;
             Configuration = configuration;
             LoggerSetup = loggerSetup;
 
+	        var hostName = Dns.GetHostName();
```

Comment:
Bad, admin may configure special stuff/ ip / etc, this just expose us on a random location